src/linalg/cholesky.rs

```diff
@@ -163,7 +170,23 @@ where
     ///
     /// Returns `None` if the input matrix is not definite-positive. The input matrix is assumed
     /// to be symmetric and only the lower-triangular part is read.
-    pub fn new(mut matrix: OMatrix<T, D, D>) -> Option<Self> {
+    pub fn new(matrix: OMatrix<T, D, D>) -> Option<Self> {
+        Self::new_with_substitute(matrix, None)
+    }
+
+    /// Attempts to compute the Cholesky decomposition of `matrix`.
+    ///
+    /// If `substitute` is given, any values during the decomposition that are 0
+    /// or otherwise fail `ComplexField::try_sqrt` will be replaced with
+    /// `substitute` instead. The `substitute` itself is subject to the same
+    /// checks and will by `try_sqrt`'d before being used.
+    ///
+    /// This allows to compensate for matrices with very small or even negative
+    /// values due to numerical errors.
+    pub fn new_with_substitute(
+        mut matrix: OMatrix<T, D, D>,
+        substitute: Option<T>,
```

Comment:
What's the case for taking `Option<T>` in the public API instead of `T`?

Reply:
There was none, except that we'd need a private method to not repeat the implementation with `new`. I added a private method now, PTAL.